### PR TITLE
Added tests for milestones/latest topic subscription.

### LINKED ISSUE #172: Client - MQTT milestones/latest subscription
## Description

Implements the `milestones/latest` event API.

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change
fixes #172 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Test case : test_milestones_latest_event test

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/client/api_events/test_milestones_latest_event.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <unity/unity.h>
+#include "client/api/events/node_event.h"
+#include "client/api/events/sub_milestone_latest.h"
+#include "events_test_config.h"
+#include "string.h"
```
Comment (oopsmonk) [inline tests/client/api_events/test_milestones_latest_event.c]: for better practice, put headers in different groups.
make sure the include syntax follow the rule: https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html

```suggestion
#include <unity/unity.h>
#include <string.h>

#include "client/api/events/node_event.h"
#include "client/api/events/sub_milestone_latest.h"
#include "events_test_config.h"
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]
please fix the following warnings:
```
[build] ../src/client/network/mqtt/mqtt_mosquitto.c:145:1: warning: non-void function does not return a value [-Wreturn-type]
[build] }
[build] ^
```
```
[build] ../src/client/api/events/node_event.c:65:37: warning: implicit conversion from enumeration type 'mqtt_event_id_t' to different enumeration type 'event_client_event_id_t' [-Wenum-conversion]
[build]       node_event->event_id = event->event_id;
[build]                            ~ ~~~~~~~^~~~~~~~
```

File: tests/client/api_events/test_milestones_latest_event.c (hunk 1 above)
Comment (rebinmannady93) [reply tests/client/api_events/test_milestones_latest_event.c]: Done.

**Comment by rebinmannady93** [comment]
> please fix the following warnings:
> 
> ```
> [build] ../src/client/network/mqtt/mqtt_mosquitto.c:145:1: warning: non-void function does not return a value [-Wreturn-type]
> [build] }
> [build] ^
> ```
> 
> ```
> [build] ../src/client/api/events/node_event.c:65:37: warning: implicit conversion from enumeration type 'mqtt_event_id_t' to different enumeration type 'event_client_event_id_t' [-Wenum-conversion]
> [build]       node_event->event_id = event->event_id;
> [build]                            ~ ~~~~~~~^~~~~~~~
> ```

Done.

File: src/client/api/events/node_event.c [hunk 2]
```diff
@@ -1,11 +1,12 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-#include "node_event.h"
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
+
 #include "client/network/mqtt/mqtt.h"
+#include "node_event.h"
```
Comment (oopsmonk) [inline src/client/api/events/node_event.c]:
```suggestion
#include "client/api/events/node_event.h"
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/client/api/events/node_event.c (hunk 2 above)
Comment (rebinmannady93) [reply src/client/api/events/node_event.c]: I had done this but when running clang-format hook, it puts node_event.h on top of node_event.h.

Comment (oopsmonk) [reply src/client/api/events/node_event.c]: > it puts node_event.h on top of node_event.h.

I'm not sure I understand what you mean by this but the point is to include the header file with the full path, the ordering shouldn't be a problem.

Comment (rebinmannady93) [reply src/client/api/events/node_event.c]: Sorry my bad, What I meant is, if I change as below :
#include "node_event.h"
#include "client/api/events/node_event.h"

and then run clang-format, it changes code to  :
#include "client/network/mqtt/mqtt.h"
#include "node_event.h"

Comment (oopsmonk) [reply src/client/api/events/node_event.c]: I think you misread the diff notation.
the suggested change is
```
#include "node_event.h"
```
to
```
#include "client/api/events/node_event.h"
```
which modify line 9 only.

> Sorry my bad, What I meant is, if I change as below :
> #include "node_event.h"
> #include "client/api/events/node_event.h"

including `node_event.h` and `client/api/events/node_event.h` looks something wrong, isn't?

Comment (rebinmannady93) [reply src/client/api/events/node_event.c]: Sorry, I misinterpreted it. Yes it should be client/api/events/node_event.h.
Done.

**Pull request merged**